### DISCUSSION TITLE
Project HÆLTH Blog #2

### DISCUSSION THREAD

**Discussion by SidMllr** [ldcdorn/haelth#29] [General]
# Project HÆLTH 🌱

Our second blog post covers the topics we worked on and goals we achieved this week.

## UI Mockup
To create a first mockup of the future user interface HÆLTH will work with, we discovered [Figma](https://www.figma.com/). In Figma, we easily visualized our ideas regarding the UI and general structure of our app. It also has great Kotlin implementation, a key point for us. On top of that, [Figma](https://www.figma.com/) provides the possibility to implement some logic behind the buttons to switch around screens. Feel free to try it out yourself, we uploaded the [.fig file](https://github.com/ldcdorn/haelth/blob/main/doc/ui_mockup.fig) to our documentary items. 
A [screenshot](https://github.com/ldcdorn/haelth/blob/main/doc/ui_mockup_overview.svg) showing all different sections is also provided:
<img width="939" alt="uimockup_overview" src="https://github.com/user-attachments/assets/5ed03dac-a6a6-4d36-84eb-25b585c70739">
As can be taken from the picture, the user starts in a login screen (this might change later, since we did not settle on the account structure yet). After that, the homescreen provides a quick overview of today's exercise, nutritional facts and cognitive exercise, so the user can easily access his preferred category. The user then can freely move around the different sections, see and edit exercises and meals or start a new meditation session (Every one of the larger columns represent one of the three sections). 

## Use case diagrams
Working on the use case diagrams, we figured that [draw.io](https://app.diagrams.net) is a great tool to create them, especially working simultaneously at the same time.
![image](https://github.com/user-attachments/assets/8452797d-fa2d-4dba-b6c4-6cf0539cd137)
For example, this diagram visualizes the use case of the user regarding his meditation exercises.
To see all of the use case diagrams and the requirements they work towards, we advise you to see the [Software Requirement Specification](https://github.com/ldcdorn/haelth/blob/main/doc/software_requirement_specification.md) in our repository.
## Software Requirement Specification
Another goal of this week was to create a "SRS". It can be found [here](https://github.com/ldcdorn/haelth/blob/main/doc/software_requirement_specification.md) in our GitHub Repository. We used some of our user stories for showing the non-functional requirements and created a few new ones, visible in the [Product Backlog](https://github.com/users/ldcdorn/projects/4). However, it focuses on the functional requirements on our project, implementing the UI Mockup and several UseCase-diagrams.

**Comment by themsgcentre** [comment]
Wir finden es beeindruckend, wie strukturiert ihr an das Projekt herangeht, besonders die detaillierte Ausarbeitung der SRS und der UI-Mockups in Figma. 
Die Aufteilung in verschiedene Kategorien wie Ernährung und kognitive Übungen finden wir sinnvoll, um das Nutzererlebnis ganzheitlich zu gestalten. 
Eine Anregung wäre, vielleicht noch die Use-Case Diagramme mit den Attributen "extend" und "include" zu versehen, wie es bei dem klassischen UML-Entwurf der Fall ist.
Insgesamt ein toller Fortschritt!

Viele Grüße,
Projekt TDC (HM)

**Comment by OrlandoLegolas** [comment]
Das klingt nach einer produktiven Woche für das HÆLTH-Projekt! Figma für die UI-Mockups zu nutzen, scheint eine gute Wahl zu sein, vor allem mit der Möglichkeit, die Navigation zwischen den Bildschirmen zu implementieren. Die klare Aufteilung in Übungen, Ernährung und Meditation auf dem Homescreen verspricht eine benutzerfreundliche Erfahrung. Ich bin gespannt, wie sich das entwickelt, wenn die Account-Struktur finalisiert ist.

Wir sind gespannt auf das nächste Update nächste Woche. 

Betse Grüße
das GymTrackerX Team
